x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/aggs/InternalInferenceAggregation.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.ml.inference.aggs;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.search.aggregations.InternalAggregation;
+import org.elasticsearch.search.aggregations.InvalidAggregationPathException;
+import org.elasticsearch.xpack.core.ml.inference.results.ClassificationInferenceResults;
+import org.elasticsearch.xpack.core.ml.inference.results.InferenceResults;
+import org.elasticsearch.xpack.core.ml.inference.results.SingleValueInferenceResults;
+import org.elasticsearch.xpack.core.ml.inference.trainedmodel.ClassificationConfig;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public class InternalInferenceAggregation extends InternalAggregation {
+
+    private final InferenceResults inferenceResult;
+
+    protected InternalInferenceAggregation(String name, Map<String, Object> metadata,
+                                           InferenceResults inferenceResult) {
+        super(name, metadata);
+        this.inferenceResult = inferenceResult;
+    }
+
+    public InternalInferenceAggregation(StreamInput in) throws IOException {
+        super(in);
+        inferenceResult = in.readNamedWriteable(InferenceResults.class);
+    }
+
+    public InferenceResults getInferenceResult() {
+        return inferenceResult;
+    }
+
+    @Override
+    protected void doWriteTo(StreamOutput out) throws IOException {
+        out.writeNamedWriteable(inferenceResult);
+    }
+
+    @Override
+    public InternalAggregation reduce(List<InternalAggregation> aggregations, ReduceContext reduceContext) {
+        throw new UnsupportedOperationException("Reducing an inference aggregation is not supported");
+    }
+
+    @Override
+    public Object getProperty(List<String> path) {
+        if (path.isEmpty()) {
+            return this;
+        } else if (path.size() == 1) {
```

Comment:
Its difficult to fully grok with all these branches. Any way to use the fact that these branches are all returning out of the function?

Something like
```
if (predicate) {
   return "foo";
}
if (predicate2) {
   return "bar";
}
return null;
```

Reply:
I've changed this again 😞  now it only respects the `value` property.

I've deleted an awful lot of branching code because as we discussed above we can come up with a better solution in a later PR